graphics/src/AssimpLoader_TEST.cc

```diff
@@ -0,0 +1,607 @@
+  ASSERT_TRUE(mat != nullptr);
+
+  // Make sure we read the specular value
+  EXPECT_EQ(math::Color(0.0, 0.0, 0.0, 1.0), mat->Ambient());
+  EXPECT_EQ(math::Color(0.64f, 0.64f, 0.64f, 1.0f), mat->Diffuse());
+  EXPECT_EQ(math::Color(0.5, 0.5, 0.5, 1.0), mat->Specular());
+  EXPECT_EQ(math::Color(0.0, 0.0, 0.0, 1.0), mat->Emissive());
+  EXPECT_DOUBLE_EQ(50.0, mat->Shininess());
+  // transparent: opaque="A_ONE", color=[1 1 1 1]
+  // transparency: 1.0
+  // resulting transparency value = (1 - color.a * transparency)
+  EXPECT_DOUBLE_EQ(0.0, mat->Transparency());
+  double srcFactor = -1;
+  double dstFactor = -1;
+  mat->BlendFactors(srcFactor, dstFactor);
+  EXPECT_DOUBLE_EQ(1.0, srcFactor);
+  EXPECT_DOUBLE_EQ(0.0, dstFactor);
+
+  common::Mesh *meshOpaque = loader.Load(
+      common::testing::TestFile("data", "box_opaque.dae"));
+  ASSERT_TRUE(meshOpaque);
+
+  EXPECT_EQ(meshOpaque->MaterialCount(), 1u);
+
+  common::MaterialPtr matOpaque = meshOpaque->MaterialByIndex(0u);
+  ASSERT_TRUE(matOpaque != nullptr);
+
+  // Make sure we read the specular value
+  EXPECT_EQ(math::Color(0.0, 0.0, 0.0, 1.0), matOpaque->Ambient());
+  EXPECT_EQ(math::Color(0.64f, 0.64f, 0.64f, 1.0f), matOpaque->Diffuse());
+  EXPECT_EQ(math::Color(0.5, 0.5, 0.5, 1.0), matOpaque->Specular());
+  EXPECT_EQ(math::Color(0.0, 0.0, 0.0, 1.0), matOpaque->Emissive());
+  EXPECT_DOUBLE_EQ(50.0, matOpaque->Shininess());
+  // transparent: opaque="A_ONE", color=[1 1 1 1]
+  // transparency: not specified, defaults to 1.0
+  // resulting transparency value = (1 - color.a * transparency)
+  EXPECT_DOUBLE_EQ(0.0, matOpaque->Transparency());
+  srcFactor = -1;
+  dstFactor = -1;
+  matOpaque->BlendFactors(srcFactor, dstFactor);
+  EXPECT_DOUBLE_EQ(1.0, srcFactor);
+  EXPECT_DOUBLE_EQ(0.0, dstFactor);
+}
+
+/////////////////////////////////////////////////
+TEST_F(AssimpLoader, ShareVertices)
+{
+  common::AssimpLoader loader;
+  common::Mesh *mesh = loader.Load(
+      common::testing::TestFile("data", "box.dae"));
+
+  // check number of shared vertices
+  std::set<unsigned int> uniqueIndices;
+  int shared = 0;
+  for (unsigned int i = 0; i < mesh->SubMeshCount(); ++i)
+  {
+    const std::shared_ptr<common::SubMesh> subMesh =
+      mesh->SubMeshByIndex(i).lock();
+    for (unsigned int j = 0; j < subMesh->IndexCount(); ++j)
+    {
+      if (uniqueIndices.find(subMesh->Index(j)) == uniqueIndices.end())
+      {
+        uniqueIndices.insert(subMesh->Index(j));
+      }
+      else
+      {
+        shared++;
+      }
+    }
+  }
+  EXPECT_EQ(shared, 12);
+  EXPECT_EQ(uniqueIndices.size(), 24u);
+
+  // check all vertices are unique
+  for (unsigned int i = 0; i < mesh->SubMeshCount(); ++i)
+  {
+    const std::shared_ptr<common::SubMesh> subMesh =
+      mesh->SubMeshByIndex(i).lock();
+    for (unsigned int j = 0; j < subMesh->VertexCount(); ++j)
+    {
+      gz::math::Vector3d v = subMesh->Vertex(j);
+      gz::math::Vector3d n = subMesh->Normal(j);
```

Comment:
Avoid using the `gz::` or `ignition::` namespaces when already using those namespaces, this helps reduce the diff between Fortress and Garden (and beyond)